### PR TITLE
[WEBUI] Avoid possibility of script in query param keys

### PR DESCRIPTION
**Pull request by srowen**
As discussed separately, this avoids the possibility of XSS on certain request param keys.

CC @vanzin

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala [hunk 1]
```diff
@@ -206,7 +206,9 @@ private[ui] class AllJobsPage(parent: JobsTab, store: AppStatusStore) extends We
       jobs: Seq[v1.JobData],
       killEnabled: Boolean): Seq[Node] = {
     // stripXSS is called to remove suspicious characters used in XSS attacks
-    val allParameters = request.getParameterMap.asScala.toMap.mapValues(_.map(UIUtils.stripXSS))
+    val allParameters = request.getParameterMap.asScala.toMap.map {
+      case (k, v) => UIUtils.stripXSS(k) -> v.map(UIUtils.stripXSS).toSeq
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala]: nit: `case (k, v) =>` should go on previous line

**Review by vanzin — COMMENTED** [review]
LGTM aside from style nit.

**Comment by srowen** [comment]
This should be OK to merge. It passed first before the last failure with no change, so the latter looks and should be unrelated

**Comment by vanzin** [comment]
Merging.

**Pull request closed without merging**